lib/http-incoming.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const originalUrl = require('original-url');
+const { URL } = require('url');
+
+const noop = str => str;
+
+const PodiumHttpIncoming = class PodiumHttpIncoming {
+    constructor(request = {}, response = {}, params = {}) {
```

Comment:
what about changing this to:

```js
constructor(request = { headers: {} }, response = {}, params = {}) {
```
to solve your headers issue?

Reply:
Its not a big deal. Just annoying and I hope get it into original-url.